### ISSUE TITLE
train co_deformable_detr_r50_1x_coco, the AP lower.

### ISSUE THREAD

**Issue by Belinda-great** [Sense-X/Co-DETR#29]
Hi Thank you for your great work!
When I used "tools/dist_train.sh projects/configs/co_deformable_detr/co_deformable_detr_r50_1x_coco.py 1 ./weight/" trained the co_deformable_detr_r50_1x_coco on coco2017, not modify the parameters, and not using the pretrain model, the AP is lower. Should I use the pretrain model for training? Or others need modify? I am looking forward to your reply.
I test the coco result:
{  Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.461
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=1000 ] = 0.633
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=1000 ] = 0.502
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=1000 ] = 0.280
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=1000 ] = 0.499
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=1000 ] = 0.604
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.649
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=300 ] = 0.649
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=1000 ] = 0.649
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=1000 ] = 0.433
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=1000 ] = 0.697 
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=1000 ] = 0.826 }
some log(loss is bigger than yours):
{"mode": "train", "epoch": 1, "iter": 50, "lr": 0.0002, "memory": 10678, "data_time": 0.07217, "enc_loss_cls": 2.1538, "enc_loss_bbox": 1.30198, "enc_loss_iou": 1.16616, "loss_cls": 2.14317, "loss_bbox": 2.60367, "loss_iou": 2.14343, "d0.loss_cls": 1.96131, "d0.loss_bbox": 2.63388, "d0.loss_iou": 2.16948, "d1.loss_cls": 2.15417, "d1.loss_bbox": 2.60833, "d1.loss_iou": 2.15319, "d2.loss_cls": 2.23642, "d2.loss_bbox": 2.60134, "d2.loss_iou": 2.14901, "d3.loss_cls": 2.1437, "d3.loss_bbox": 2.60013, "d3.loss_iou": 2.14763, "d4.loss_cls": 2.11218, "d4.loss_bbox": 2.60227, "d4.loss_iou": 2.14508, "loss_rpn_cls": 3.50659, "loss_rpn_bbox": 0.47681, "loss_cls0": 5.12987, "acc0": 94.27344, "loss_bbox0": 1.34195, "loss_cls1": 14.23468, "loss_bbox1": 13.64706, "loss_centerness1": 7.91569, "loss_cls_aux0": 1.76192, "loss_bbox_aux0": 1.0423, "loss_iou_aux0": 0.56317, "d0.loss_cls_aux0": 1.65817, "d0.loss_bbox_aux0": 1.03151, "d0.loss_iou_aux0": 0.51077, "d1.loss_cls_aux0": 1.65367, "d1.loss_bbox_aux0": 1.03382, "d1.loss_iou_aux0": 0.52373, "d2.loss_cls_aux0": 1.72817, "d2.loss_bbox_aux0": 1.03599, "d2.loss_iou_aux0": 0.53485, "d3.loss_cls_aux0": 1.75864, "d3.loss_bbox_aux0": 1.0379, "d3.loss_iou_aux0": 0.5451, "d4.loss_cls_aux0": 1.75284, "d4.loss_bbox_aux0": 1.03995, "d4.loss_iou_aux0": 0.55428, "loss_cls_aux1": 1.79561, "loss_bbox_aux1": 1.18594, "loss_iou_aux1": 1.14411, "d0.loss_cls_aux1": 1.69971, "d0.loss_bbox_aux1": 1.18362, "d0.loss_iou_aux1": 1.14339, "d1.loss_cls_aux1": 1.71459, "d1.loss_bbox_aux1": 1.18391, "d1.loss_iou_aux1": 1.14331, "d2.loss_cls_aux1": 1.76991, "d2.loss_bbox_aux1": 1.18443, "d2.loss_iou_aux1": 1.14341, "d3.loss_cls_aux1": 1.80116, "d3.loss_bbox_aux1": 1.18488, "d3.loss_iou_aux1": 1.14361, "d4.loss_cls_aux1": 1.78536, "d4.loss_bbox_aux1": 1.18529, "d4.loss_iou_aux1": 1.14385, "loss": 136.48589, "grad_norm": 158.88891, "time": 0.77338}
{"mode": "train", "epoch": 1, "iter": 100, "lr": 0.0002, "memory": 10678, "data_time": 0.00637, "enc_loss_cls": 1.76474, "enc_loss_bbox": 1.1683, "enc_loss_iou": 1.16472, "loss_cls": 2.05801, "loss_bbox": 2.06883, "loss_iou": 1.80247, "d0.loss_cls": 1.88228, "d0.loss_bbox": 2.09778, "d0.loss_iou": 1.81527, "d1.loss_cls": 1.93532, "d1.loss_bbox": 2.0773, "d1.loss_iou": 1.79806, "d2.loss_cls": 1.96658, "d2.loss_bbox": 2.06948, "d2.loss_iou": 1.80279, "d3.loss_cls": 2.03127, "d3.loss_bbox": 2.06957, "d3.loss_iou": 1.80236, "d4.loss_cls": 2.05364, "d4.loss_bbox": 2.06872, "d4.loss_iou": 1.80287, "loss_rpn_cls": 2.22799, "loss_rpn_bbox": 0.37238, "loss_cls0": 3.29344, "acc0": 96.75781, "loss_bbox0": 1.24619, "loss_cls1": 13.36668, "loss_bbox1": 13.64313, "loss_centerness1": 7.90339, "loss_cls_aux0": 1.54687, "loss_bbox_aux0": 1.09403, "loss_iou_aux0": 0.55488, "d0.loss_cls_aux0": 1.37439, "d0.loss_bbox_aux0": 1.09063, "d0.loss_iou_aux0": 0.53468, "d1.loss_cls_aux0": 1.39558, "d1.loss_bbox_aux0": 1.0916, "d1.loss_iou_aux0": 0.54084, "d2.loss_cls_aux0": 1.44337, "d2.loss_bbox_aux0": 1.09214, "d2.loss_iou_aux0": 0.54455, "d3.loss_cls_aux0": 1.49814, "d3.loss_bbox_aux0": 1.09262, "d3.loss_iou_aux0": 0.54755, "d4.loss_cls_aux0": 1.52944, "d4.loss_bbox_aux0": 1.09319, "d4.loss_iou_aux0": 0.55106, "loss_cls_aux1": 1.59381, "loss_bbox_aux1": 1.14691, "loss_iou_aux1": 1.17268, "d0.loss_cls_aux1": 1.40565, "d0.loss_bbox_aux1": 1.14653, "d0.loss_iou_aux1": 1.17178, "d1.loss_cls_aux1": 1.44506, "d1.loss_bbox_aux1": 1.14654, "d1.loss_iou_aux1": 1.17201, "d2.loss_cls_aux1": 1.4978, "d2.loss_bbox_aux1": 1.14658, "d2.loss_iou_aux1": 1.17212, "d3.loss_cls_aux1": 1.55235, "d3.loss_bbox_aux1": 1.14668, "d3.loss_iou_aux1": 1.17229, "d4.loss_cls_aux1": 1.57807, "d4.loss_bbox_aux1": 1.14692, "d4.loss_iou_aux1": 1.1726, "loss": 122.95549, "grad_norm": 107.07448, "time": 0.70219}

**Comment by TempleX98** [comment]
I will check it, do you use imagenet pretrained r50?

**Comment by Belinda-great** [comment]
> I will check it, do you use imagenet pretrained r50?
Thank you for your feedback!
No, I havn't using any models for pretained.
Should I use imagenet pretrained r50?  Could you sent me a link?

**Comment by TempleX98** [comment]
The code `init_cfg=dict(type='Pretrained', checkpoint='torchvision://resnet50')` in r50 configs means we use the resnet50 weights from torchvision

**Comment by Belinda-great** [comment]
> The code `init_cfg=dict(type='Pretrained', checkpoint='torchvision://resnet50')` in r50 configs means we use the resnet50 weights from torchvision
Yes, I didn't modify the config "co_deformable_detr_r50_1x_coco.py", so configs means we use the resnet50 weights.
I used cuda number one train it, Will it make a difference?

**Comment by Belinda-great** [comment]
[20230803_035648.log](https://github.com/Sense-X/Co-DETR/files/12310773/20230803_035648.log)
This is my log, please help me check there has any question. Thank you very much.

**Comment by TempleX98** [comment]
It seems you only use 1 GPU for training. We use a batch size of 16 (2 images per GPU).
